plugins/utils/neoclip.nix

```
@@ -0,0 +1,240 @@
+      Although, in practice no real slowdowns were noticed.
+      Alternatively see `db_pull` and `db_push` functions in the
+      [README](https://github.com/AckslD/nvim-neoclip.lua#custom-actions).
+    '';
+
+    db_path =
+      helpers.defaultNullOpts.mkStr { __raw = "vim.fn.stdpath('data') .. '/databases/neoclip.sqlite3'"; }
+        ''
+          The path to the sqlite database to store history if `enable_persistent_history=true`.
+          Defaults to `$XDG_DATA_HOME/nvim/databases/neoclip.sqlite3`.
+        '';
+
+    filter = helpers.defaultNullOpts.mkLuaFn null ''
+      A function to filter what entries to store (default all are stored).
+      This function filter should return `true` (include the yanked entry) or `false` (don't include
+      it) based on a table as the only argument, which has the following keys:
+
+      - `event`: The event from `TextYankPost` (see ``:help TextYankPost` for which keys it contains).
+      - `filetype`: The filetype of the buffer where the yank happened.
+      - `buffer_name`: The name of the buffer where the yank happened.
+    '';
+
+    preview = helpers.defaultNullOpts.mkBool true ''
+      Whether to show a preview (default) of the current entry or not.
+      Useful for for example multiline yanks.
+      When yanking the filetype is recorded in order to enable correct syntax highlighting in the
+      preview.
+
+      NOTE: in order to use the dynamic title showing the type of content and number of lines you
+      need to configure `telescope` with the `dynamic_preview_title = true` option.
+    '';
+
+    prompt = helpers.defaultNullOpts.mkStr null ''
+      The prompt string used by the picker (`telescope`/`fzf-lua`).
+    '';
+
+    default_register =
+      helpers.defaultNullOpts.mkNullable (with helpers.nixvimTypes; either str (listOf str)) "\""
+        ''
+          What register to use by default when not specified (e.g. `Telescope neoclip`).
+          Can be a string such as `"\""` (single register) or a table of strings such as
+          `["\"" "+" "*"]`.
+        '';
+
+    default_register_macros = helpers.defaultNullOpts.mkStr "q" ''
+      What register to use for macros by default when not specified (e.g. `Telescope macroscope`).
+    '';
+
+    enable_macro_history = helpers.defaultNullOpts.mkBool true ''
+      If true (default) any recorded macro will be saved, see
+      [macros](https://github.com/AckslD/nvim-neoclip.lua#macros).
+    '';
+
+    content_spec_column = helpers.defaultNullOpts.mkBool false ''
+      Can be set to `true` to use instead of the preview.
+    '';
+
+    disable_keycodes_parsing = helpers.defaultNullOpts.mkBool false ''
+      If set to true, macroscope will display the internal byte representation, instead of a proper
+      string that can be used in a map.
+      So a macro like `"one<CR>two"` will be displayed as `"one\ntwo"`.
+      It will only show the type and number of lines next to the first line of the entry.
```

```suggestion
      If set to true, macroscope will display the internal byte representation, instead of a proper
      string that can be used in a map.

      So a macro like `"one<CR>two"` will be displayed as `"one\ntwo"`.
      It will only show the type and number of lines next to the first line of the entry.
```

Sorry to nit, I just think more paragraphs usually makes the docs much more readable.

---

No no, you are right.